test/integration/puppetlabs/services/master/environment_classes_int_test.clj

```diff
@@ -111,4 +188,262 @@
                                                   (str
                                                    "{ foo_1 => 1, "
                                                    "foo_2 => 2 }")}]}]}]}
-                (response->class-info-map response))))))))
+                (response->class-info-map response))
+             "unexpected body following code change")))
+     (testing "environment class cache invalidation for one environment"
```

Comment:
This test ended up being pretty long.  The `testing` blocks prior to this one have order dependencies - one test must run before the result via caching can be validated in the next one.  These last two tests around cache invalidation could be promoted into top-level testing blocks but we'd have to bring the full puppetserver stack up and down two more times in doing so.  I opted for wrapping these all together under one puppetserver service stack instance to reduce the performance overhead a bit.  If others think it would be a better trade-off to separate these tests out, I'd be open to that, though.